bottlecap/src/bin/bottlecap/main.rs

```diff
@@ -106,15 +105,35 @@ async fn next_event(client: &reqwest::Client, ext_id: &str) -> Result<NextEventR
     let base_url = base_url(EXTENSION_ROUTE)
         .map_err(|e| Error::new(std::io::ErrorKind::InvalidData, e.to_string()))?;
     let url = format!("{base_url}/event/next");
-    client
+
+    let response = client
         .get(&url)
         .header(EXTENSION_ID_HEADER, ext_id)
         .send()
         .await
-        .map_err(|e| Error::new(std::io::ErrorKind::InvalidData, e.to_string()))?
-        .json()
-        .await
-        .map_err(|e| Error::new(std::io::ErrorKind::InvalidData, e.to_string()))
+        .map_err(|e| {
+            debug!("Request failed: {}", e);
+            Error::new(std::io::ErrorKind::InvalidData, e.to_string())
+        })?;
+
+    let status = response.status();
+    let text = response.text().await.map_err(|e| {
+        debug!("Failed to read response body: {}", e);
+        Error::new(std::io::ErrorKind::InvalidData, e.to_string())
+    })?;
+
+    if !status.is_success() {
+        debug!("HTTP Error {} - Response: {}", status, text);
```

Comment:
Should we be more explicit with this errors?

To mention something about the Next Event?

Reply:
yeah and they probably should be errors as we can't do anything if we get a failure here